### PR TITLE
chore: add main cors entry

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Add `main` deploy preview pattern to CORS. (ie: https://639aee43101e366250794f27--regen-registry.netlify.app/)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] targeted `dev` branch
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: server/Server.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ const REGISTRY_REDWOOD_HOSTNAME_PATTERN =
 const REGISTRY_HAMBACH_HOSTNAME_PATTERN =
   /hambach--regen-registry\.netlify\.app$/;
 const REGISTRY_V4_HOSTNAME_PATTERN = /v4--regen-registry\.netlify\.app$/;
+const MAIN_PREVIEW_HOSTNAME_PATTERN = /\d+--regen-registry\.netlify\.app$/;
```
Comment (blushi) [inline server/Server.ts:37]: to account for non digit characters in the preview hash
```suggestion
const MAIN_PREVIEW_HOSTNAME_PATTERN = /[a-z0-9]+--regen-registry\.netlify\.app$/;
```

Comment (flagrede) [reply server/Server.ts:37]: Seems like this part has only numbers but let's play it safe you're right.

Comment (blushi) [reply server/Server.ts]: in https://639aee43101e366250794f27--regen-registry.netlify.app/, 639aee43101e366250794f27 does have some letters

**Review by blushi — APPROVED** [review]
pre-approving, pending lint issue

**Review by wgwz — APPROVED** [review]

**Pull request merged**